src/opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -86,24 +86,65 @@ void ExtNetwork::add_branch(Branch branch)
     this->m_branches.push_back( std::move(branch) );
 }
 
-void ExtNetwork::drop_branch(const std::string& uptree_node, const std::string& downtree_node) {
-    auto downtree_branches = this->downtree_branches( downtree_node );
-    if (downtree_branches.empty()) {
+void ExtNetwork::add_or_replace_branch(Branch branch)
+{
+    const std::string& uptree_node = branch.uptree_node();
+    const std::string& downtree_node = branch.downtree_node();
+
+    // Add any missing nodes
+    for (const std::string& nodename : { downtree_node, uptree_node }) {
+        if (!this->has_node(nodename)) {
+            this->m_nodes.insert_or_assign(nodename, Node{nodename});
+        }
+        if (!this->has_indexed_node_name(nodename)) {
+            this->add_indexed_node_name(nodename);
+        }
+    }
+    
+    // Remove any other branch uptree from downtree_node (gathering tree structure required)
+    auto uptree_link = this->uptree_branch( downtree_node );
+    if (uptree_link.has_value()){
+        const auto& old_uptree_node = uptree_link.value().uptree_node();
+        this->drop_branch(old_uptree_node, downtree_node);
+    }
+    
+    // Update existing branch if it exists
+    auto downtree_branches = this->downtree_branches( uptree_node );
+    if (!downtree_branches.empty()) {
         auto branch_iter = std::find_if(this->m_branches.begin(), this->m_branches.end(), [&uptree_node, &downtree_node](const Branch& b) { return (b.uptree_node() == uptree_node && b.downtree_node() == downtree_node); });
-        if (branch_iter != this->m_branches.end())
-            this->m_branches.erase( branch_iter );
+        if (branch_iter != this->m_branches.end()) {
+            *branch_iter = branch;
+            return;
+        }
+    }
+    
+    this->m_branches.push_back( std::move(branch) );
+}
+
+bool ExtNetwork::is_disconnected(const std::string& node_name) const {
```

Comment:
I do not think this function is used in this PR.  Is this function needed for other PR?

Reply:
You're right , it is not used now. It was used in an early version, and kept for possible use with some consistency checking at a later stage, but can remove it if that is preferable.

Reply:
Yes, Please remove it if it is not used in this PR. It is easy to introduce it again with the PR that actually uses the function.

Reply:
Will do.

Reply:
And also, it is a very good idea to have a function to check the consistency of the network. With this function, we make it clear what types of network structure is allowed.

Reply:
I agree, but suggest to make a new PR for that after the current set of PRs are done.